statsmodels/sandbox/regression/penalized.py

```diff
@@ -82,52 +228,97 @@ def fit(self, lambd=1.):
         y = self.wendog[:,None]
         #why are sigma2_e * lambd multiplied, not ratio?
         #larger lambd -> stronger prior  (it's not the variance)
+        # Bayesian: lambd is precision = 1/sigma2_prior
         #print('lambd inside fit', lambd
-        xpx = np.dot(x.T, x) + \
+        xx = np.dot(x.T, x)
+        xpx = xx + \
               sigma2_e * lambd * np.dot(r_matrix.T, np.dot(sigma_prior_inv, r_matrix))
         xpy = np.dot(x.T, y) + \
               sigma2_e * lambd * np.dot(r_matrix.T, np.dot(sigma_prior_inv, q_matrix))
         #xpy = xpy[:,None]
 
-        xpxi = np.linalg.pinv(xpx)
+        xpxi = np.linalg.pinv(xpx, rcond=1e-15**2)  #to match pinv(x) in OLS case
         params = np.dot(xpxi, xpy)    #or solve
         params = np.squeeze(params)
-        self.normalized_cov_params = xpxi    #why attach it to self, i.e. model?
+        # normalized_cov_params should have sandwich form xpxi @ xx @ xpxi
+        if cov_type == 'sandwich':
+            self.normalized_cov_params = xpxi.dot(xx).dot(xpxi)
+        elif cov_type == 'data-prior':
+            self.normalized_cov_params = xpxi    #why attach it to self, i.e. model?
```

Comment:
see code comment
attached to model, but result gets xpxi

Reply:
note normalized_cov_params depends on fit argument.
i.e. model attribute becomes "stale"